### ISSUE TITLE
Investigate adding axe-core to test suit

### ISSUE THREAD

**Issue by djbuckley** [manchesergit/material-ui#154]
investigate adding [axe-core](https://github.com/dequelabs/axe-core) to the test suit
add notes here as to what would changes would be required to ensure that all components are covered
test should not include any theme's i.e. contrast ratio's, as they can be decided by the designer of the UI that is using this framework

tests should be on individual components

once a procedure has been identified, then a wiki page should be written describing the process and this issue closed

**Comment by davidjacob93** [comment]
First use the command npm install axe-core --save-dev on the root directory of the project. Then download the module and add it to the dependency list.

**Comment by davidjacob93** [comment]
https://www.sitepoint.com/automated-accessibility-checking-with-axe/

this site gives help on how to use axe-core

**Comment by davidjacob93** [comment]
axe.run(function (err, results) {
  if (results.violations.length) {
    console.warn(results.violations);
  }
});

Adding this axe core code to the testing script will log all of the errors to the browser console

**Comment by djbuckley** [comment]
issue #214 cannot begin until this has been clompleted

**Comment by davidjacob93** [comment]
Here it goes ...

To check for accessibility violations in a webpage, we need a couple of new modules to be added to the devdependencies in the package.json file.

In the material-ui folder open a command prompt. Run these commands.

`npm install phantomjs-prebuilt selenium-webdriver --save-dev`
`npm install axe-core axe-webdriverjs --save-dev`

After this installation has been completed, click on on start menu and type in 'Environment Variables' and press enter. Then select 'PATH' and click on the edit button. Copy the path where phantomjs.exe file is and append it in the PATH box. Click OK and OK again.

Now open any text editor (preferably Atom) and open the Material-ui project folder which you would have cloned from git.

On any of the components you would like to run the test in the src directory, right click on and click on 'New File' and name it axe.js. In that file copy and paste the code below.

```
const axeBuilder = require('axe-webdriverjs');
const webDriver = require('selenium-webdriver');

const driver = new webDriver.Builder()
  .forBrowser('phantomjs')
  .build();

driver
  .get('http://localhost:3000/#/components/*COMPONENTNAME*')
  .then(() => {
    axeBuilder(driver)
      .analyze((results) => {
        console.log(results);
      });
  });

```
Save this file.

Now go on package.json file in the directory and change the default test script entry;
```
"scripts": {
"test":"node src/*COMPONENTNAME*/axe.js"
}
```
Save this file.

Now open a command prompt inside the material-ui folder and type 'npm test'. Wait few seconds and you should see a list of violations and passes that axe found on the terminal.

It should look something like this.

![capture](https://user-images.githubusercontent.com/25061353/28527930-0bd33a18-7084-11e7-968f-f318b37ec4e8.PNG)

**Comment by djbuckley** [comment]
we need to be able to run this without having to run a server up to host the entire package

we should be able to run the axe tests on each component without ever having to run npm start
